iam.yml

```diff
@@ -10,13 +10,138 @@ Parameters:
+                  # TODO: limit this
+                  - 's3:*'
+                Resource: 'arn:aws:s3:::cdo-*javabuilder*-content/*'
+              # Read from Javabuilder SQS Queue
+              - Effect: Allow
+                Action:
+                  - "sqs:ReceiveMessage"
+                  - "sqs:DeleteMessage"
+                  - "sqs:GetQueueAttributes"
+                  - "sqs:GetQueueUrl"
+                  - "sqs:ListDeadLetterSourceQueues"
+                  - "sqs:ListQueues"
+                Resource:
+                  # TODO: limit this
+                  - '*' 
+              - Effect: Allow
+                Action:
+                  - "execute-api:Invoke"
+                  - "execute-api:ManageConnections"
+                Resource: !Sub "arn:aws:execute-api:*:${AWS::AccountId}:${JavabuilderApiId}/*/$connect"
+  SessionManagerMessageRelayLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: ['sts:AssumeRole']
+            Effect: Allow
+            Principal: {Service: [lambda.amazonaws.com]}
+      Policies:
+        - PolicyName: SessionManagerMessageRelayLambdaRole
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "sqs:ReceiveMessage"
+                  - "sqs:DeleteMessage"
+                  - "sqs:GetQueueAttributes"
+                  # TODO: limit this
+                  - "sqs:*"
+                Resource: '*'
+              # Allow invoking of the Build & Run Lambda
+              - Effect: Allow
+                Action:
+                  - "lambda:InvokeFunction"
+                Resource:
+                  # TODO: limit this
+                  - "*"
+  AuthorizerLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: ['sts:AssumeRole']
+            Effect: Allow
+            Principal: {Service: [lambda.amazonaws.com]}
+      Policies:
+        - PolicyName: AuthorizerLambdaExecutionPolicy
+          PolicyDocument:
+            Statement:
+              # Authorizer lambdas need to create, read, and update dynamodb records for throttling.
+              - Effect: Allow
+                Action:
+                  - 'dynamodb:GetItem'
+                  - 'dynamodb:PutItem'
+                  - 'dynamodb:UpdateItem'
+                  - 'dynamodb:Query'
+                Resource:
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_tokens"
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_user_requests"
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_teacher_associated_requests"
+                  - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*_blocked_users"
+              # Allow invoking of the Build & Run Lambda
```

Comment:
They shouldn't need to invoke the build & run lambda. I'm not sure how API Gateway sets this up, but they might need to be able to invoke the StartSession(AndRelayMessage) lambda after authorizing the payload, but I think API Gateway is what actually invokes that lambda.

Reply:
I don't see anything in the code for authorizer other than the dynamodb `put` and `update`, so I think I can scale down to just dynamodb permissions.